virtcontainers/pkg/intel/govmm/ich/ich.go

```diff
@@ -0,0 +1,443 @@
+	// VFIO is a direct assigned PCI device or PCI VF
+	VFIO NetDeviceType = "VFIO"
+
+	// VHOSTUSER is a vhost-user port (socket)
+	VHOSTUSER NetDeviceType = "vhostuser"
+)
+
+type Memory struct {
+	// Size is the amount of memory made available to the guest.
+	// It should be suffixed with M or G for sizes in megabytes or
+	// gigabytes respectively.
+	Size string
+	// Path is the file path of the memory device. It points to a local
+	// file path used by FileBackedMem.
+	Path string
+}
+
+type VCPU struct {
+	
+	Size uint32
+	
+}
+
+// Kernel is the guest kernel configuration structure.
+type Kernel struct {
+	// Path is the guest kernel path on the host filesystem.
+	Path string
+
+	// Params is the kernel parameters string.
+	Params string
+}
+
+// Object is a ich object representation.
+type Object struct {
+
+	// ID is the user defined object ID.
+	ID string
+
+}
+
+// Valid returns true if the Object structure is valid and complete.
+func (object Object) Valid() bool {
+
+	return true
+}
+
+// QemuParams returns the qemu parameters built out of this Object device.
+func (object Object) IchParams(config *Config) []string {
+
+	var ichParams []string
+
+	// TODO
+	
+	return ichParams
+}
+
+
+/************************************************************************************
+ *
+ * NetDeviceType is a ich networking device type.
+ *
+*************************************************************************************/ 
+type NetDeviceType string
+
+type NetDevice struct {
+	// Type is the netdev type (e.g. tap).
+	Type NetDeviceType
+
+	// ID is the netdevice identifier.
+	ID string
+
+	// IfName is the interface name,
+	IFName string
+
+	// VHost enables virtio device emulation from the host kernel instead of from qemu.
+	VHost bool
+
+	// MACAddress is the networking device interface MAC address.
+	MACAddress string
+
+
+}
+
+// IchParams returns the ich parameters built out of this network device.
+func (netdev NetDevice) IchParams(config *Config) []string {
+	
+	var ichParams []string
+
+	// TODO more then 1 netdev needs improvements.
+
+	ichParams = append(ichParams, "--net", fmt.Sprintf("tap=%s,mac=%s", netdev.IFName, netdev.MACAddress))
+
+	return ichParams
+}
+
+// Valid returns true if the NetDevice structure is valid and complete.
+func (netdev NetDevice) Valid() bool {
+	if netdev.ID == "" || netdev.IFName == "" {
+		return false
+	}
+
+	switch netdev.Type {
+	case TAP:
+		return true
+	default:
+		return false
+	}
+}
+
+/************************************************************************************
+ *
+ * DiskDevice represents the path to the (root) disk image
+ *
+*************************************************************************************/  
+type DiskDevice struct {
+
+	Path string
+
+}
+
+// IchParams returns the ich parameters built out of this network device.
+func (diskdev DiskDevice) IchParams(config *Config) []string {
+	
+	var ichParams []string
+
+	ichParams = append(ichParams, "--disk", diskdev.Path)
```

Comment:
I was testing cloud hypervisor, I got an error the most recent code expects `--disk path=diskdev.Path` after that I was able to boot a container \o/

Reply:
Changed! I'm indeed a bit behind pulling the latest cloudhypervisor. I'll build a recent one to check this.